distsys/resources/relaxedmailboxes.go

```diff
@@ -13,14 +13,21 @@ import (
 )
 
 // RelaxedMailboxesMaker produces a distsys.ArchetypeResourceMaker for a
-// collection of TCP mailboxes. However, relaxed mailboxes don't follow 2PC
-// semantics strictly same as TCP mailboxes. The main difference is that
-// when a critical section successfully sends a message using relaxed
-// mailboxes (res.Write returns with no error), it will be not possible
-// to abort that critical section anymore. Therefore, it's not always safe
-// to use relaxed mailboxes instead of TCP mailboxes. It's only safe to use
-// them when there is at most one network send operation and all other
-// operations are guaranteed to commit successfully.
+// collection of TCP mailboxes. It has the same guaranttees as tcp mailboxes,
+// however, relaxed mailboxes don't follow 2PC semantics strictly same
+// as TCP mailboxes. The main difference is that when a critical section
+// successfully sends a message using relaxed remotes mailboxes (res.Write
+// returns with no error), it will be not possible to abort that critical
+// section anymore. Therefore, it's not always safe to use relaxed mailboxes
+// instead of TCP mailboxes. It's only safe to use them in a critical section
+// when there is at most one network send operation in the it and all
+// succeeding operations in the critical section are guaranteed to commit
+// successfully. Also with relaxed mailboxes, it's not safe have an await
+// statement after a network send in a critical section.
+// Note that we only the remove rollback support in the relaxed mailboxes and
+// don't remove the timeout support. Reading from a relaxed local mailbox might
+// timeout and it's OK. Also writing to a relaxed remote mailbox might timeout
+// and it's fine too.
```

Comment:
Last nitpick:

That's why, with these restrictions, it is still possible to use a limited form of `either` statement, as long as `await` comes before the network write, and timing out on a network write is sequentially the last reason the `either` branch might fail.